Organize imports

mauriciovieira opened the pull request:
1. remove unused imports
2. organize imports (sort and unify comments)
3. remove unused eslint-disable

closes #2107

mauriciovieira commented:
@apinf/developers please review

mauriciovieira commented:
Notice I standardized the first comments with:

```
// Meteor packages imports

// Npm packages imports

// Collection imports

// APINF imports
```

The imports follow this order unless breaks eslint

brylie commented:
I am wondering whether removing all of these explicit imports is the 'right thing to do'. In other words, it may be useful, and at some point necessary, to explicitly import Meteor components. Consistency with imports also helps our developers to better understand how things are connected, and is a common practice with other frameworks that don't have such a 'magical' environment as Meteor.

brylie commented:
The Meteor Guide also recommends using explicit imports for Meteor libraries.

mauriciovieira commented:
What I mind mostly is the consistency. Before we started to run eslint (a month ago) we had meteor imports only in some files.

brylie commented:
Yes, for the same reason we had lint in only a few files. In effect, the strict import has not been a problem for the duration of our development, and has only recently been caught by the linter. Hence, the work in progress towards consistent use of explicit imports.

mauriciovieira commented:
No problem, I'll write some script to put them back (removing using sed was easy). 
I spent a few hours yesterday sorting and commenting all files and won't like to waste this work.

brylie commented:
Cool. I am not trying to give you grief. I'm just clarifying a direction we are heading, based on the recommendation of the Meteor Guide.

mauriciovieira commented:
Use node to reinsert the import statements http://nicolashery.com/parse-data-files-using-nodejs-streams/ (https://github.com/nicolashery/example-stream-parser/blob/master/lib/parser.js)

brylie commented:
Is this ready for review? It might be easier to just create a new branch, since most of the changes were unnecessary. Also, I am not sure we should add 'meteor' to the eslint environment, since it might discourage developers from being explicit about Meteor imports.

mauriciovieira commented:
I prefer not undoing the manual work I've done to comment and sort all imports.

I wanted to work on a script to automatically fix the meteor no-undef. This script might be useful. I am still working on that.

mauriciovieira commented:
@brylie I rebased the commits. 

I standardized the first comments with:

```
// Meteor packages imports

// Meteor contributed packages imports

// Npm packages imports

// Collection imports

// APINF imports
```

Also, I noticed some broken references in the develop branch and fixed them. 

Please review.

mauriciovieira commented:
'Add meteor env to eslintrc' is not a suitable description for this work. It's better to call it 'Organize imports'. I renamed it

marla-singer commented on api_docs/client/swaggerUI/swaggerUiContent/swaggerUiContent.js:
```diff
@@ -1,14 +1,17 @@
 
-import Apis from '/apis/collection';
+// Meteor contributed packages imports
 import ApiKeys from '/api_keys/collection';
```
Hm. Looks like it's `// Collection imports` not `// Meteor contributed packages imports`

mauriciovieira replied:
🎉

marla-singer commented on api_docs/files/collection/index.js:
```diff
@@ -1,3 +1,4 @@
+// Meteor packages imports
 import { FileCollection } from 'meteor/vsivsi:file-collection';
```
`Meteor contributed packages imports` here

marla-singer commented on api_keys/collection/schema.js:
```diff
@@ -1,5 +1,7 @@
 import { SimpleSchema } from 'meteor/aldeed:simple-schema';
 
+// Npm packages imports
 import ApiKeys from './';
```
`Collection imports` here

mauriciovieira replied:
👍

marla-singer commented on apis/client/profile/intro/intro.js:
```diff
@@ -1,11 +1,10 @@
+import { Template } from 'meteor/templating';
 
 // Meteor imports
 import { TAPi18n } from 'meteor/tap:i18n';
```
`Meteor contributed packages imports` here

mauriciovieira replied:
👍

marla-singer commented on apis/client/profile/settings/details/autoform.js:
```diff
@@ -1,7 +1,7 @@
-// Meteor package import
+// Meteor packages imports
 import { AutoForm } from 'meteor/aldeed:autoform';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/client/profile/status/convert_status_code.js:
```diff
@@ -1,4 +1,4 @@
-// Meteor packages import
+// Meteor packages imports
 import { TAPi18n } from 'meteor/tap:i18n';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/collection/schema.js:
```diff
@@ -1,4 +1,7 @@
+// Meteor packages imports
 import { SimpleSchema } from 'meteor/aldeed:simple-schema';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/collection/server/permissions.js:
```diff
@@ -1,7 +1,10 @@
-import _ from 'lodash';
-
+// Meteor packages imports
 import { Roles } from 'meteor/alanning:roles';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/flags/collection/permissions.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Roles } from 'meteor/alanning:roles';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/logo/collection/index.js:
```diff
@@ -1,3 +1,4 @@
+// Meteor packages imports
 import { FileCollection } from 'meteor/vsivsi:file-collection';
```
`Meteor contributed packages imports` here

marla-singer commented on apis/server/methods/status.js:
```diff
@@ -1,5 +1,7 @@
+import { check } from 'meteor/check';
+
 
 import Apis from '/apis/collection';
```
`Collection imports` here

mauriciovieira replied:
🎉

marla-singer commented on users/collection/users_permissions.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Meteor } from 'meteor/meteor';
 
+
```
New line

mauriciovieira replied:
👍

marla-singer commented on backlog/collection/index.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Mongo } from 'meteor/mongo';
 
+
```
Excessive new line

marla-singer commented on core/client/error_templates/go_back_button/goBackButton.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Template } from 'meteor/templating';
 
+
```
Excessive new line

mauriciovieira replied:
👍

marla-singer commented on dashboard/client/charts/table/table.js:
```diff
@@ -1,5 +1,7 @@
 import { ReactiveVar } from 'meteor/reactive-var';
+import { Template } from 'meteor/templating';
+
 
```
Excessive new line

mauriciovieira replied:
👍

marla-singer commented on feedback/client/list/list.js:
```diff
@@ -1,7 +1,9 @@
 
+import { Modal } from 'meteor/peppelg:bootstrap-3-modal';
+
 import Feedback from '../../collection';
```
`Collection imports` here

mauriciovieira replied:
👍

marla-singer commented on feedback/collection/schema.js:
```diff
@@ -1,4 +1,6 @@
 import { Meteor } from 'meteor/meteor';
+
 import { SimpleSchema } from 'meteor/aldeed:simple-schema';
 import Feedback from './';
```
`Collection imports` here

mauriciovieira replied:
🎉

marla-singer commented on home/client/footer/homeFooter.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Template } from 'meteor/templating';
 
+
```
Excessive new line

brylie replied:
:boom:

mauriciovieira replied:
✌️

marla-singer commented on proxy_backends/client/modal_form/remove_selected_proxy/remove_selected_proxy.js:
```diff
@@ -0,0 +1,28 @@
+// Meteor packages imports
+import { Template } from 'meteor/templating';
```
@mauriciovieira `removed_selected_proxy` folder was deleted

mauriciovieira replied:
Oh, thanks for catching this :-)

marla-singer commented on status/client/status.js:
```diff
@@ -1,6 +1,8 @@
 import { Meteor } from 'meteor/meteor';
 import { Template } from 'meteor/templating';
 
+
```
Excessive new line

brylie replied:
:hammer:

mauriciovieira replied:
:1st_place_medal:

marla-singer commented on status/server/status.js:
```diff
@@ -1,5 +1,7 @@
+// Meteor packages imports
 import { Meteor } from 'meteor/meteor';
 
+
```
Excessive new line

brylie replied:
:last_quarter_moon_with_face:

mauriciovieira replied:
👀

marla-singer commented on users/client/profile/profile.js:
```diff
@@ -1,6 +1,8 @@
 import { Meteor } from 'meteor/meteor';
 import { Template } from 'meteor/templating';
 
+
```
Excessive new line

brylie replied:
:pager:

mauriciovieira replied:
👀

mauriciovieira commented:
@marla-singer Thank you so much for catching these errors. 
Please review again.
